backend/app/core/exceptions.py

```
@@ -25,28 +25,46 @@ def __init__(self, message: str, *, status_code: int | None = None, error_code:
+class ServiceUnavailableError(AppError):
+    """Exception for unavailable required infrastructure (503)."""
+
+    status_code = status.HTTP_503_SERVICE_UNAVAILABLE
+    error_code = "service_unavailable"
+
+
 def register_exception_handlers(app: FastAPI) -> None:
+    """Register custom exception handlers for the FastAPI application."""
+
     @app.exception_handler(AppError)
     async def handle_app_error(request: Request, exc: AppError) -> JSONResponse:
```

Ruff (0.16.3)

[warning] Unused function argument: `request`

(ARG001)